presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoDriverUri.java

```diff
@@ -265,6 +268,15 @@ public void setupClient(OkHttpClient.Builder builder)
                 }
                 builder.addInterceptor(tokenAuth(ACCESS_TOKEN.getValue(properties).get()));
             }
+            if (EXTERNAL_AUTHENTICATION.getValue(properties).orElse(false)) {
+                if (!useSecureConnection) {
+                    throw new SQLException("Authentication using external authorization requires SSL to be enabled");
+                }
+                setupExternalAuthenticator(
```

Comment:
This method is here as it's going to be used in cli as well. It tries to follow a convention established earlier.

Reply:
Ok so the memoize and Supplier are no go I think. This is a hack for now - I admit. The issue here is that we need to create a separate client, with probably some different characteristics, that will still be sharing some of the configuration - like the ssl one. Still by using a memoized Supplier as you suggest, we would end up with a OkHttp client that has also the ExternalAuthenticator attached to it, just a different one - I don't like this duality. In general, refactoring around it would require quite a few changes and I don't want to put it in this pr.

Reply:
> This method is here as it's going to be used in cli as well. 

Let's do not predict the future when not needed.. It is easy to extract common code when we would make support CLI. Now it obscures.

Reply:
> Still by using a memoized Supplier as you suggest, we would end up with a OkHttp client that has also the ExternalAuthenticator attached to it, just a different one

Please add a comment. This is one argument to inline this method as it do not hide the fact that builder is used twice.